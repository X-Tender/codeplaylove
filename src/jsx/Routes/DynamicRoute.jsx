import { PureComponent } from 'react';

export default class DynamicRoute extends PureComponent {
	state = {
		component: null,
	};
	componentWillMount() {
		this.props.load().then(component => {
			this.setState(() => ({
				component: component.default ? component.default : component,
			}));
		});
	}
	render() {
		return this.props.children(this.state.component);
	}
}
