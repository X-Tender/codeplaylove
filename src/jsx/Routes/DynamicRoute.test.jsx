import React from 'react';
import { shallow } from 'enzyme';
import DynamicRoute from './DynamicRoute';

describe('DynamicRoute', () => {
	const children = jest.fn();

	describe('component not loaded', () => {
		const loadWithoutComponent = () =>
			new Promise(resolve => {
				resolve({ default: null });
			});

		const comp = shallow(<DynamicRoute load={loadWithoutComponent}>{children}</DynamicRoute>);

		it('renders correctly', () => {
			expect(comp).toMatchSnapshot();
		});
	});

	describe('component loaded', () => {
		const loadWithComponent = () =>
			new Promise(resolve => {
				resolve({ default: <div /> });
			});

		const comp = shallow(<DynamicRoute load={loadWithComponent}>{children}</DynamicRoute>);

		it('renders correctly', () => {
			expect(comp).toMatchSnapshot();
		});
	});
});
