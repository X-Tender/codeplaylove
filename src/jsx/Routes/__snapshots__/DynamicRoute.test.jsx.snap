// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicRoute component loaded renders correctly 1`] = `""`;

exports[`DynamicRoute component not loaded renders correctly 1`] = `""`;
